appinventor/blocklyeditor/src/blocklyeditor.js

```diff
@@ -40,6 +40,31 @@ Blockly.configForTypeBlock = {
 Blockly.BlocklyEditor.render = function() {
 };
 
+function unboundVariableHandler(myBlock, yailText) {
+  var unbound_vars = Blockly.LexicalVariable.freeVariables(myBlock);
+  unbound_vars = unbound_vars.toList();
+  if (unbound_vars.length == 0) {
+    Blockly.ReplMgr.putYail(yailText, myBlock);
+  } else {
+    var form = "<form>" + Blockly.Msg.DIALOG_ENTER_VALUES + "<br>";
+    for (var v in unbound_vars) {
+      form  +=  unbound_vars[v] + '<input type=text name=' + unbound_vars[v] + '><br>';
```

Comment:
One recommendation here is to include either a colon or an equal sign between the name of the variable and the textbox to provide some context. I'm inclined to use an equal sign like most programming languages use for assignment.